R/downloadableReactTable.R

```diff
@@ -0,0 +1,166 @@
+#'
+#' downloadableReactTable module is extending \code{?reactable} package table functions by creating
+#' a custom high-functionality table paired with \link[periscope2]{downloadFile} button.
+#' The table has search and highlight functionality, infinite scrolling, sorting by columns and
+#' returns a reactive dataset of selected items.
+#'
+#' \link[periscope2]{downloadFile} button will be hidden if \code{downloadableReactTableUI} parameter
+#' \code{downloadtypes} is empty
+#'
+#' @param id character id for the object
+#' @param downloadtypes vector of values for data download types
+#' @param hovertext download button tooltip hover text
+#' @param contentHeight viewable height of the table (any valid css size value)
+#' @param singleSelect whether the table should only allow a single row to be
+#' selected at a time (FALSE by default allows multi-select).
+#'
+#' @return list of downloadFileButton UI and reactable table and hidden inputs for contentHeight and singleSelect options
+#'
+#' @section Table Features:
+#' \itemize{
+#'     \item Consistent styling of the table
+#'     \item downloadFile module button functionality built-in to the table (it will be shown only if downloadtypes is defined)
+#'     \item Ability to show different data from the download data
+#'     \item Table is automatically fit to the window size with infinite
+#'     y-scrolling
+#'     \item Table search functionality including highlighting built-in
+#'     \item Multi-select built in, including reactive feedback on which table
+#'     items are selected
+#' }
+#'
+#' @section Example:
+#' \code{downloadableReactTableUI("mytableID", c("csv", "tsv"),
+#' "Click Here", "300px")}
+#'
+#' @section Notes:
+#' When there are no rows to download in any of the linked downloaddatafxns the
+#' button will be hidden as there is nothing to download.
+#'
+#' @section Shiny Usage:
+#' Call this function at the place in ui.R where the table should be placed.
+#'
+#' Paired with a call to \code{downloadableReactTable(id, ...)}
+#' in server.R
+#'
+#' @seealso \link[periscope2]{downloadableReactTable}
+#' @seealso \link[periscope2]{downloadableTableUI}
+#' @seealso \link[periscope2]{downloadableTable}
+#' @seealso \link[periscope2]{downloadFile}
+#' @seealso \link[periscope2]{logViewerOutput}
+#' @seealso \link[periscope2]{downloadFile}
+#' @seealso \link[periscope2]{downloadFile_ValidateTypes}
+#' @seealso \link[periscope2]{downloadFile_AvailableTypes}
+#' @seealso \link[periscope2]{downloadablePlot}
+#'
+#' @examples
+#' if (interactive()) {
+#'  library(shiny)
+#'  library(periscope2)
+#'  shinyApp(ui = fluidPage(fluidRow(column(width = 12,
+#'    downloadableReactTableUI("object_id1",
+#'                             downloadtypes = c("csv", "tsv"),
+#'                             hovertext     = "Download the data here!",
+#'                             contentHeight = "300px",
+#'                             singleSelect  = FALSE)))),
```

Comment:
this is the default value, correct?  There is no need to specify this if it is the default.  For the example if you want to show it, put it as TRUE.